### PR TITLE
gnrc_ipv6_netif: don't advertise routers without GUAs

### PR DESCRIPTION
**Pull request by authmillenon**
Introduces behavior proposed in https://github.com/RIOT-OS/RIOT/pull/4174#issuecomment-151621768

### CODE REVIEW

**Comment by cgundogan** [comment]
just to be sure that I do not overlook something here: in this PR `global_prefix_set` is never set to true, right?
Will this be added in follow-ups?

**Comment by cgundogan** [comment]
ah, I know read the comment you put there. sorry (:

File: sys/net/gnrc/network_layer/ipv6/netif/gnrc_ipv6_netif.c [hunk 1]
```diff
@@ -793,6 +793,9 @@ void gnrc_ipv6_netif_init_by_dev(void)
         eui64_t iid;
         uint16_t tmp;
         gnrc_ipv6_netif_t *ipv6_if = gnrc_ipv6_netif_get(ifs[i]);
+        /* cppcheck-suppress variableScope confusion through ifdefs */
+        bool global_prefix_set = false; /* XXX should be set to true if there ever is an
+                                         * auto-configured prefix set */
```
Comment (cgundogan) [inline sys/net/gnrc/network_layer/ipv6/netif/gnrc_ipv6_netif.c]: what's an auto-configured prefix? do you mean auto-configured address?

Comment (authmillenon) [inline sys/net/gnrc/network_layer/ipv6/netif/gnrc_ipv6_netif.c]: A prefix that is configured in this function.

**Comment by authmillenon** [comment]
Ping?

**Comment by OlegHahm** [comment]
Have you tested? If I'm not mistaken I can still see RAs even if no prefix is configured on the 6LBR.

**Comment by authmillenon** [comment]
I have exactly the opposite problem on naticve. The router isn't sending RAs even if there is a GUA configured :/.

**Comment by authmillenon** [comment]
@OlegHahm can you test your configuration again. I think I fixed that now. Mine is just the fact that the RTR_ADV flag is not set. Will add capability to the shell command for that.

**Comment by authmillenon** [comment]
See https://github.com/RIOT-OS/RIOT/pull/4303 for the flag issue

**Comment by authmillenon** [comment]
Rebased to current master.

File: sys/net/gnrc/network_layer/ipv6/netif/gnrc_ipv6_netif.c (hunk 1 above)
Comment (OlegHahm) [inline sys/net/gnrc/network_layer/ipv6/netif/gnrc_ipv6_netif.c]: Where is this variable ever set?

**Comment by OlegHahm** [comment]
ACK, please squash.

**Comment by authmillenon** [comment]
Squashed

**Comment by authmillenon** [comment]
Fixed hanging problem

**Comment by authmillenon** [comment]
Rebased

**Comment by authmillenon** [comment]
Rebased due to #3309

**Comment by OlegHahm** [comment]
Eh, have you tested again?

**Comment by authmillenon** [comment]
Args, no :(

**Pull request merged**